src/main/java/org/betonquest/betonquest/Instruction.java

```diff
@@ -639,4 +632,110 @@ public PartParseException(final String message, final Throwable cause) {
             super("Error while parsing " + (lastOptional == null ? currentIndex : lastOptional + " optional") + " argument: " + message, cause);
         }
     }
+
+    public static class InstructionParser {
+
+        private final Set<Character> STRING_DELIMITERS = Set.of('"','\'', '`');
+        private static final Pattern KEY_PATTERN = Pattern.compile("^[a-zA-Z]+$");
+        private final Deque<ArrayList<String>> args = new ArrayDeque<>();
+        private final StringBuilder sequence = new StringBuilder();
+        private boolean insideQuote = false;
+        private final String input;
+
+        public InstructionParser(final String input) {
+            this.input = input.trim();
```

Comment:
I am not sure if we want to trim the input. The reason are the notify and command events.